compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -534,5 +534,21 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
         sym.addAnnotation(Annotation(defn.ExperimentalAnnot, sym.span))
         sym.companionModule.addAnnotation(Annotation(defn.ExperimentalAnnot, sym.span))
 
+    private def scala2LibPatch(tree: TypeDef)(using Context) =
+      val sym = tree.symbol
+      if ctx.settings.Yscala2Stdlib.value
+        && sym.isClass && sym.is(Module) && !sym.derivesFrom(defn.SerializableClass)
+        && sym.companionClass.derivesFrom(defn.SerializableClass)
+      then
+        // Add Serializable to companion objects of serializable classes
+        sym.info = sym.info match
```

Comment:
Setting `sym.info` like this does not seem right to me. Normally changes to the `info` of symbols must be done in an `InfoTransform`.

Reply:
I removed the info update. It should be fine not to patch those infos while compiling the stdlib as we do not really use those parents. We also only pickle the tree, which means that the unpickled info will have the correct info.

Reply:
Er, that seems worse to me. Now the symbol table and the trees are not consistent with each other. That's something that should turn up in `Ycheck`.